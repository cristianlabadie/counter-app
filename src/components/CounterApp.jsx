import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  let [counter, setCounter] = useState(value);
  const handleIncrement = () => setCounter(counter + 1);
  const handleDecrement = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="test-counter">{counter}</h2>

      <button onClick={handleIncrement}>+</button>
      <button onClick={handleDecrement}>-</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
