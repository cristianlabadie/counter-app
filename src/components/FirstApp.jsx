import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h2 data-testid="test-title"> {title} </h2>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};

FirstApp.defaultProps = {
  subtitle: "Soy un subtitulo",
};
