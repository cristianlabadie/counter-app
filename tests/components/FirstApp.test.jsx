import { render } from "@testing-library/react";
import { FirstApp } from "../../src/components/FirstApp";

describe("Pruebas en <FirstApp />", () => {
  test("debe de hacer match con el snapshot", () => {
    const title = "Estoy creando una app en react";
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el titulo en un h1"', () => {
    const title = "Estoy creando una app en react";
    const { container, getByText, getByTestId } = render(
      <FirstApp title={title} />,
    );

    expect(getByText(title)).toBeTruthy();

    // Esto es demasiado especifico y no es recomendable
    // const h1 = container.querySelector("h1");
    // expect(h1.innerHTML).toContain(title);

    // Esto es mejor
    expect(getByTestId("test-title").innerHTML).toContain(title); // con esto nos aseguramos que el contenido sea igual, independientemente de si es un h1, h2, etc.

    // con esto nos aseguramos que el contenido sea estrictamente en el contenido, si posee espacios, también los considerará
    // expect(getByTestId("test-title").innerHTML).toBe(title);
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    const title = "Estoy creando una app en react";
    const subtitle = "Esto es un subtitulo";
    const { getByText } = render(
      <FirstApp title={title} subtitle={subtitle} />,
    );

    expect(getByText(subtitle)).toBeTruthy();
  });
});
