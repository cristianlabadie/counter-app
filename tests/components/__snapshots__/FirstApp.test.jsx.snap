// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h2
    data-testid="test-title"
  >
     
    Estoy creando una app en react
     
  </h2>
  <p
    data-testid="test-subtitle"
  >
    Soy un subtitulo
  </p>
  <p />
</div>
`;
